pkg/virt-controller/watch/vmi.go

```diff
@@ -729,15 +730,17 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		}
 
 	case vmi.IsScheduled():
-		// Nothing here
+		if !vmiPodExists {
+			vmiCopy.Status.Phase = virtv1.Failed
+		}
 		break
 	default:
 		return fmt.Errorf("unknown vmi phase %v", vmi.Status.Phase)
 	}
 
 	// VMI is owned by virt-handler, so patch instead of update
 	if vmi.IsRunning() || vmi.IsScheduled() {
-		patchBytes, err := prepareVMIPatch(vmi, vmiCopy)
+		patchBytes, err := prepareVMIPatch(vmi, vmiCopy, pod)
```

Comment:
it is more accurate to pass in `vmiPodExists` here rather than `pod`.

Reply:
Done.